pkg/pipeline/common_steps.go

```diff
@@ -69,29 +74,49 @@ func addTenantLabel(obj PipelineObject, data *ExecutionContext) ExecutionResult
 
 func updateObject(obj PipelineObject, data *ExecutionContext) ExecutionResult {
 
-	oldResourceVersion := obj.GetObjectMeta().GetResourceVersion()
-
 	rtObj, ok := obj.(runtime.Object)
 	if !ok {
-		return ExecutionResult{
-			Abort: true,
-			Err:   fmt.Errorf("object ist not a valid runtime.object: %v", obj.GetObjectMeta().GetName()),
+		return ExecutionResult{Err: errors.New("object is not a runtime object")}
+	}
+
+	if !specAndMetaEqual(obj, data.originalObject) {
+		err := data.Client.Update(context.TODO(), rtObj)
+		if err != nil {
+			if k8serrors.IsConflict(err) {
+				return ExecutionResult{Abort: true}
```

Comment:
Wouldn't it be better to also return the error here, to make sure it's immediately being re-queued?

Reply:
If that error happens it should already be re-queued due to the pending change? If we return the error it will be logged, which is the intention to avoid with this `if` statemenet.

Reply:
Not necessarily. How about we check the returned error for `IsConflict` globally and just return `requeue: true` but no error?

Reply:
The pipeline execution currently only passes through the error back to the controller. But we can catch the `IsConflict` in the controller itself and set the requeue.